patches/minecraft/net/minecraft/tileentity/TileEntityPiston.java.patch

```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/tileentity/TileEntityPiston.java
++++ ../src-work/minecraft/net/minecraft/tileentity/TileEntityPiston.java
+@@ -145,7 +145,7 @@
+ 
+             if (!list1.isEmpty())
+             {
+-                boolean flag = this.field_174932_a.func_177230_c() == Blocks.field_180399_cE;
++                boolean flag = this.field_174932_a.func_177230_c().isStickyBlock(this.field_174932_a, this.field_145850_b, this.field_174879_c);
```

Comment:
I'm not entirely sure about this, it doesn't look like the state is actually at the given position in the world at this point in time. It's at least worth noting this in the docs for `isStickyBlock` in any case.